### DISCUSSION TITLE
snmpwalks works but snmp_exporter complains with error in unmarshalResponse: error decoding value: error parsing OID Value: invalid OID length

### DISCUSSION THREAD

**Discussion by vcarceler** [prometheus/snmp_exporter#1099] [Q&A]
We have a switch D-Link DGS-1100-24pv2.

snmpwalk works and scrapes the device in 3m.

snmp_exporter shows this error:

```
An error has occurred while serving metrics:

error collecting metric Desc{fqName: "snmp_error", help: "Error scraping target", constLabels: {module="if_mib"}, variableLabels: {}}: error walking target 192.168.240.54: error in unmarshalResponse: error decoding value: error parsing OID Value: invalid OID length

```

Any advice?

**Comment by tanxiao1990** [comment]
same issue , how can i fix it

**Comment by ralphjin** [comment]
https://github.com/gosnmp/gosnmp/blame/v1.38.0/gosnmp.go#L24

	// MaxOids is the maximum number of OIDs permitted in a single call,
	// otherwise error. MaxOids too high can cause remote devices to fail
	// strangely. 60 seems to be a common value that works, but you will want
	// to change this in the GoSNMP struct
	MaxOids = 60

it seems that it is a limitation in gosnmp. is there any way to increase this max value?

**Reply by SuperQ** [reply]
I don't think that's related. The error returned `invalid OID length` is the result of a [zero-length byte sequence](https://github.com/gosnmp/gosnmp/blob/1100b9e1add22664b90f056b607720e0f7f1d15e/helper.go#L649-L652).
